velox/serializers/tests/UnsafeRowSerializerTest.cpp

```diff
@@ -73,10 +74,102 @@ class UnsafeRowSerializerTest : public ::testing::Test {
     test::assertEqualVectors(deserialized, rowVector);
   }
 
+  void
+  testSerialize(RowVectorPtr rowVector, int8_t* expectedData, size_t dataSize) {
+    std::ostringstream out;
+    serialize(rowVector, &out);
+    EXPECT_EQ(std::memcmp(expectedData, out.str().data(), dataSize), 0);
+  }
+
+  void
+  testDeserialize(int8_t* data, size_t dataSize, RowVectorPtr expectedVector) {
+    auto results = deserialize(
+        asRowType(expectedVector->type()),
+        std::string_view(reinterpret_cast<const char*>(data), dataSize));
+    test::assertEqualVectors(expectedVector, results);
+  }
+
   std::shared_ptr<memory::MemoryPool> pool_;
   std::unique_ptr<VectorSerde> serde_;
 };
 
+// These expected binary buffers were samples taken using Spark's java code.
+TEST_F(UnsafeRowSerializerTest, tinyint) {
+  int8_t data[20] = {0, 0, 0,   16, 0, 0, 0, 0, 0, 0,
+                     0, 0, 123, 0,  0, 0, 0, 0, 0, 0};
+  auto expected = makeRowVector({makeFlatVector(std::vector<int8_t>{123})});
+
+  testSerialize(expected, data, 20);
+  testDeserialize(data, 20, expected);
+}
+
+TEST_F(UnsafeRowSerializerTest, bigint) {
+  int8_t data[20] = {0, 0, 0,  16, 0,   0,   0, 0, 0, 0,
+                     0, 0, 62, 28, -36, -33, 2, 0, 0, 0};
+  auto expected =
+      makeRowVector({makeFlatVector(std::vector<int64_t>{12345678910})});
+
+  testSerialize(expected, data, 20);
+  testDeserialize(data, 20, expected);
+}
+
+TEST_F(UnsafeRowSerializerTest, double) {
+  int8_t data[20] = {0, 0, 0,   16, 0,  0,  0,   0,  0,    0,
+                     0, 0, 125, 63, 53, 94, -70, 73, -109, 64};
+  auto expected =
+      makeRowVector({makeFlatVector(std::vector<double>{1234.432})});
+
+  testSerialize(expected, data, 20);
+  testDeserialize(data, 20, expected);
+}
+
+TEST_F(UnsafeRowSerializerTest, boolean) {
+  int8_t data[20] = {0, 0, 0, 16, 0, 0, 0, 0, 0, 0,
+                     0, 0, 1, 0,  0, 0, 0, 0, 0, 0};
+  auto expected = makeRowVector({makeFlatVector(std::vector<bool>{true})});
+
+  testSerialize(expected, data, 20);
+  testDeserialize(data, 20, expected);
+}
+
+TEST_F(UnsafeRowSerializerTest, string) {
```

Comment:
Should these tests go into velox/row/tests/UnsafeRowSerdeTest.cpp ?

Reply:
These test the full "package" or "stream" (the size, plus potentially many unsafe rows). The ones under velox/row only test the content of a single unsafe row.